Add Project API

rafael-azevedo opened the pull request:
This PR adds the project API.  To do so I moved common types between the project and projectClaim types to a common files. I also added updated examples of both types.

rafael-azevedo commented:
/hold

rafael-azevedo commented:
When testing this implementation I realized that Kube has a default type called projects. While this implementation does work you can get the object by saying `oc get projects.gcp.managed.openshift.io` we many want to rethink the name of this api for more clarity .

NautiluX commented:
I'm in favor of renaming. GCPProject maybe?

jhernand commented:
@tzvatot please review.

NautiluX commented on deploy/crds/gcp_v1alpha1_project_crd.yaml:
```diff
@@ -0,0 +1,115 @@
+    name: ClaimName
+    type: string
+  - JSONPath: .spec.projectClaimCRLink.namespace
+    description: Namesspace of corresponding project CR
```
should this be project claim CR?

NautiluX commented on pkg/apis/gcp/v1alpha1/common_types.go:
```diff
@@ -0,0 +1,13 @@
+	ID   string `json:"id"`
+}
+
+// NamespacedName contains the name of a secret and its namespace
```
We use this as generic name of any Kubernetes object, right? Like, as a reference to a `ProjectClaim`, so I assume it's not just secret.

NautiluX commented on pkg/apis/gcp/v1alpha1/project_types.go:
```diff
@@ -0,0 +1,91 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Shouldn't we remove this comment?

rafael-azevedo replied:
We have left this in on other repos, but I don't see why we shouldn't remove it.  Do you think we should just remove it ?

NautiluX replied:
I understand it as note by framework for us to remove once we understood this file is for us to touch ;)

rafael-azevedo replied:
I will remove on all files in this operator

rafael-azevedo commented:
Updated the type to name `GCPProject` and changed the `GCPProjectName` field in both api types to `GCPProjectID` for better ease of reading. The `GCPProjectID` is the name of the Project have have instantiated in the GCP cloud and not the CR name.
